tmt/base.py

```diff
@@ -496,6 +497,21 @@ def __init__(self, node, run=None):
             (key, str(value)) for key, value
             in node.get('environment', dict()).items()])
 
+        # Environment variables from files
+        env_files = self.node.get("environment-file") or []
+        assert isinstance(env_files, list), f"environment_file parameter should be a list. " \
+                                            f"Received {type(env_files)}"
+        env_files_vars = tmt.utils.environment_file_to_dict(env_files)
+        assert not set(env_files_vars).intersection(set(self._environment)), (
```

Comment:
Should raise `tmt.utils.GeneralError` instead (see first comment about asserts)

Reply:
okay